app/domain/audits/save_audit.rb

```diff
@@ -0,0 +1,54 @@
+module Audits
+  class SaveAudit
+    def self.call(*args)
+      new(*args).call
+    end
+
+    attr_accessor :attributes, :audit, :content_item, :next_content_item, :success, :user
+
+    def initialize(attributes:, content_id:, filter:, user_uid:)
+      self.attributes = attributes.to_h
+      self.content_item = Content::Item.find_by!(content_id: content_id)
+      self.audit = Audits::Audit.find_or_initialize_by(content_item: content_item)
+      self.next_content_item = FindNextItem.call(content_item, filter)
+      self.user = User.find_by(uid: user_uid)
+    end
+
+    def call
+      audit.content_item = content_item
+      audit.user = user
+
+      clear_redirect_urls
```

Comment:
We don't clear the fields always, so I would try to be more clear on these methods. When you read the method you could be tricked.

Reply:
Good point - thanks. Have renamed https://github.com/alphagov/content-performance-manager/pull/346/commits/bdfe179db8ee7d7f72ca0dbc282ffced02da9f34#diff-bcf8bfffc9defab9b861ec826b66da7aR43 but feels clunky.